fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/RecalculateInterestPoster.java

```diff
@@ -0,0 +1,152 @@
+                        numberOfRetries = maxNumberOfRetries + 1;
+                    } catch (CannotAcquireLockException
+                            | ObjectOptimisticLockingFailureException exception) {
+                        logger.info("Recalulate interest job has been retried  "
+                                + numberOfRetries + " time(s)");
+                        /***
+                         * Fail if the transaction has been retired for
+                         * maxNumberOfRetries
+                         **/
+                        if (numberOfRetries >= maxNumberOfRetries) {
+                            logger.warn("Recalulate interest job has been retried for the max allowed attempts of "
+                                    + numberOfRetries
+                                    + " and will be rolled back");
+                            sb.append("Recalulate interest job has been retried for the max allowed attempts of "
+                                    + numberOfRetries
+                                    + " and will be rolled back");
+                            break;
+                        }
+                        /***
+                         * Else sleep for a random time (between 1 to 10
+                         * seconds) and continue
+                         **/
+                        try {
+                            Random random = new Random();
+                            int randomNum = random
+                                    .nextInt(maxIntervalBetweenRetries + 1);
+                            Thread.sleep(1000 + (randomNum * 1000));
+                            numberOfRetries = numberOfRetries + 1;
+                        } catch (InterruptedException e) {
+                            sb.append("Interest recalculation for loans failed " + exception.getMessage()) ;
+                            break;
+                        }
+                    } catch (Exception e) {
+                        Throwable realCause = e;
+                        if (e.getCause() != null) {
+                            realCause = e.getCause();
+                        }
+                        logger.error("Interest recalculation for loans failed for account:"	+ loanId + " with message "
+                                + realCause.getMessage());
+                        sb.append("Interest recalculation for loans failed for account:").append(loanId).append(" with message ")
+                                .append(realCause.getMessage());
+                        numberOfRetries = maxNumberOfRetries + 1;
+                    }
+                    i++;
+                }
+                logger.info("Loans count " + i);
+            }
+            if (sb.length() > 0) {
+                try {
+                    throw new JobExecutionException(sb.toString());
+                } catch (JobExecutionException e) {
+                    logger.info("JobExecutionException occured :"+e.toString());
```

Comment:
use `,e` instead of `e.toString()` here, so:

    logger.info("JobExecutionException occured :", e);

also, unless there is a good reason which I'm missing, don't you want to use `logger.error` instead of `logger.info` in this case?

Reply:
@wkk91193 could you _Resolve conversation_ yourself whenever you amend PRs and have taken review feedback into account - this makes it easier for reviewers? You appear to have addressed this point - will you click _Resolve conversation_ here?